src/components/LogView/LogLineText.js

```diff
@@ -16,11 +16,51 @@ type Props = {
   port: ?string,
   text: string,
   highlightText: string[],
-  handleDoubleClick: ()=> void,
+  handleDoubleClick: () => void,
+  prettyPrint: boolean,
+}
+
+export function findJSONObjectsInLine(text: string) {
+  var startIndex = 0;
+  var numBraces = 0;
+  var chunks = [];
+  for (var i = 0; i < text.length; i++) {
+    if (text[i] === '{') {
+      if (numBraces === 0 && i !== 0) {
+        const lineBreak = (startIndex === 0) ? '' : '\n';
+        chunks.push(lineBreak + text.substring(startIndex, i));
+        startIndex = i;
+      }
+      numBraces++;
+    } else if (text[i] === '}') {
+      numBraces--;
+      if (numBraces === 0) {
+        try {
+          const lineBreak = (startIndex === 0) ? '' : '\n';
+          const jsonObj = JSON.parse(text.substring(startIndex, i + 1));
+          const formattedString = lineBreak + JSON.stringify(jsonObj, null, 2).replace(/"([^"]+)":/g, '$1:');
+          chunks.push(formattedString);
+          startIndex = i + 1;
+          console.log('this time it works!')
+        } catch (e) {
+          console.log('this always happens on first attempt to parse, for some reason');
+        }
+      }
+    }
+  }
+  if (startIndex !== text.length) {
+    chunks.push('\n' + text.substring(startIndex));
+  }
+  return chunks;
 }
 
 export default class LogLineText extends React.PureComponent<Props> {
   lineRef: ?HTMLSpanElement = null;
+  prettyPrintedText: ?string[] = [];
+
+  componentDidMount() {
```

Comment:
I kind of expected to see this in componentDidUpdate. Could you double check that we remount the component whenever you change the parent state? Probably nothing needs to change since it works, just curious

Reply:
I put it in componentDidMount because in my head, this should only run once, and I didn't want it to be called every time we re-render.